src/shared/install.c

```diff
@@ -2426,32 +2457,75 @@ int unit_file_query_preset(UnitFileScope scope, const char *root_dir, const char
                                 continue;
 
                         parameter = first_word(l, "enable");
-                        if (parameter) {
-                                if (fnmatch(parameter, name, FNM_NOESCAPE) == 0) {
-                                        log_debug("Preset file says enable %s.", name);
-                                        return 1;
-                                }
-
-                                continue;
-                        }
+                        if (parameter)
+                                rule = (PresetRule) {
+                                        .pattern = strdup(parameter),
```

Comment:
What do you think about tossing that under https://github.com/systemd/systemd/pull/3176/files#diff-c8c27856003270b2ac8678c07bc91ab8R2473.

```c
2473                         if (rule.action) {
2474                                 if (!rule.parameter)
2475                                         return -ENOMEM;
2476 
2477                                 if (!GREEDY_REALLOC(ps.rules, n_allocated, ps.n_rules + 1))
2478                                         return -ENOMEM;
```